Kommunicate/Classes/KMConversationViewController.swift

```diff
@@ -166,6 +166,36 @@ open class KMConversationViewController: ALKConversationViewController {
         }
     }
 
+    open override func addMessagesToList(_ messageList: [Any]) {
+        guard let messages = messageList as? [ALMessage] else { return }
+        var filteredArray = [ALMessage]()
+        let contactService = ALContactService()
+        for message in messages {
+            if viewModel.channelKey != nil, viewModel.channelKey == message.groupId {
+                let delayInterval = KMAppUserDefaultHandler.shared.botMessageDelayInterval
+                if delayInterval > 0 {
+                    let alContact = contactService.loadContact(byKey: "userId", value:  message.to)
+                    if alContact?.roleType == NSNumber.init(value: AL_BOT.rawValue) {
+                        updateTyingStatus(status: true, userId: message.to)
```

Comment:
Also, we are calling this multiple times in the main thread, and that means if the interval is above 2-3 secs, then the typing indicator will just be shown once for all bot messages. Each bot message is delayed, but the typing indicator's timing is not associated with each message.